core/server/src/main/java/alluxio/master/file/meta/InodeLockList.java

```diff
@@ -60,9 +80,27 @@ public synchronized void unlockLast() {
    * Locks the given inode in write mode, and adds it to this lock list.
    *
    * @param inode the inode to lock
+   * @param parent the expected parent inode
+   * @param name the expected name of the inode to be locked
+   * @throws InvalidPathException if the inode is no long consistent with the caller's expectations
+   */
+  public synchronized void lockWriteAndCheckFullPath(Inode<?> inode, Inode parent, String name)
+      throws InvalidPathException {
+    inode.lockWriteAndCheckFullPath(parent, name);
+    mInodes.add(inode);
+    mLockModes.add(InodeTree.LockMode.WRITE);
+  }
+
+  /**
+   * Locks the given inode in write mode, and adds it to this lock list.
+   *
+   * @param inode the inode to lock
+   * @param parent the expected parent inode
+   * @throws InvalidPathException if the inode is no long consistent with the caller's expectations
    */
-  public synchronized void lockWrite(Inode<?> inode) {
-    inode.lockWrite();
+  public synchronized void lockWriteAndCheckParent(Inode<?> inode, Inode parent)
```

Comment:
I previously wanted to respect alphabetical order for the sake of consistency, but I agree it is better to put the method with fewer arguments first despite not being an overloaded method.